CoreRemoting/RemotingSession.cs

```diff
@@ -310,7 +310,7 @@ private void ProcessGoodbyeMessage(WireMessage request)
             
             _rawMessageTransport.SendMessage(_server.Serializer.Serialize(resultMessage));
             
-            _server.SessionRepository.RemoveSession(_sessionId);
+            Task.Run(() => _server.SessionRepository.RemoveSession(_sessionId));
```

Comment:
I'm a bit worried about potential exceptions here.
`RemoveSession` calls `session.Dispose()` which can throw, at least in theory.
Unawaited `Task.Run` here would swallow the exception and it would remain unhandled.
What do you think?

Reply:
By design, Dispose methods should work without exceptions, otherwise the resources they need to release will leak out. If we call this task with expectation, then _currentlyProcessedMessagesCounter.Signal() will not be called in line 273, which we expect in line 738 and a deadlock will occur. The same approach is used in the Close() method on line 686.

Reply:
>By design, Dispose methods should work without exceptions

Totally agree with that, throwing Dispose is an anti-pattern. But in reality, it may throw :)

>_currentlyProcessedMessagesCounter.Signal() will not be called in line 273

Yes, I understand why it's done this way, I just wonder is there any way to dispatch the exception somewhere.

>The same approach is used in the Close() method on line 686.

Ok :)

Reply:
We can call OnErrorOccured like in line 269